test/test_checks.py

```diff
@@ -9,6 +24,30 @@ def test_get_noqa_lines(self):
         self.assertEqual(checker.get_noqa_lines(checker.get_file_contents()), [2])
 
 
+class TestFlake8Stdin(TestCase):
+
+    @expectedFailureIf(not FLAKE8_2)
+    def test_stdin(self):
+        """Test using stdin."""
+        filename = get_absolute_path('data/doubles.py')
+        env = os.environ.copy()
+        env['PYTHONIOENCODING'] = 'utf8'
```

Comment:
Well I dislike undefined behavior, but it should be in theory possible to just work on bytes instead of strings.